src/fsharp/TypeChecker.fs

```diff
@@ -4224,10 +4224,55 @@ let rec TcTyparConstraint ridx cenv newOk checkCxs occ (env: TcEnv) tpenv c =
     | WhereTyparSubtypeOfType(tp, ty, m) ->
         let ty', tpenv = TcTypeAndRecover cenv newOk checkCxs ItemOccurence.UseInType env tpenv ty
         let tp', tpenv = TcTypar cenv env newOk tpenv tp
-        if newOk = NoNewTypars && isSealedTy cenv.g ty' then 
-            errorR(Error(FSComp.SR.tcInvalidConstraintTypeSealed(), m))
-        AddCxTypeMustSubsumeType ContextInfo.NoContext env.DisplayEnv cenv.css m NoTrace ty' (mkTyparTy tp') 
+        let tryGetInlineInterface =
+            match ty' with
```

Comment:
You should always use the helpers to deconstruct, especially `tryDestAppTy` in this case. This look through type inference variables and type abbreviations